scripts/mod_loader/modui/dialog_helper.lua

```diff
@@ -136,25 +136,36 @@ end
 
 -- //////////////////////////////////////////////////////////////////////
 
-function sdlext.buildSimpleDialog(title, w, h)
-	local frame = Ui()
-			:widthpx(w):heightpx(h)
-			:decorate({ DecoFrameHeader(), DecoFrame() })
-			:caption(title)
+function sdlext.buildSimpleDialog(title, options)
+	if options then
+		Assert.Equals("table", type(options))
+	else
+		options = {}
+	end
+
+	local maxW = options.maxW or 0.5 * ScreenSizeX()
+	local maxH = options.maxH or 0.5 * ScreenSizeY()
+
+	local frame = UiWeightLayout()
+		:widthpx(maxW):heightpx(maxH)
+		:vgap(0)
+		:orientation(false)
+		:decorate({ DecoFrameHeader(), DecoFrame() })
+		:caption(title)
 
 	frame.scroll = UiScrollArea()
-			:width(1):height(1)
-			:padding(10)
-			:addTo(frame)
+		:width(1):height(1)
+		:padding(10)
```

Comment:
Maybe it would be an idea to have an option to define padding when creating the dialog?
Padding cannot realistically be changed later, since other elements are based on this value.